query/src/interpreters/interpreter_grant_privilege.rs

```diff
@@ -51,6 +73,40 @@ impl Interpreter for GrantPrivilegeInterpreter {
         _input_stream: Option<SendableDataBlockStream>,
     ) -> Result<SendableDataBlockStream> {
         let plan = self.plan.clone();
+
+        // *.`table` is not allowed
```

Comment:
case by case match is more complex, but not quite sure if there is a regular check for the pattern?

Reply:
Or we can first to check database_pattern and then table_pattern

Reply:
it seems we can seperate the grant object (global, database, or table) in the parsing phase

Reply:
how about adding a GrantObject enum on this?

it'd has three hands:

- Global
- Database(String) 
- Table(Option<String>, String) // if the database is not specified, taking the current db in session

Reply:
@BohuTANG I've refactored out an GrantObject enum, and checked `*.table` as invalid syntax in the parsing phase

please help review again when you get time 🤝